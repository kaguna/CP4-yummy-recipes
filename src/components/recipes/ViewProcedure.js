/**
 * This shows the procedure to prepare the recipe.
 */
import React, { Component } from 'react';

class Viewprocedure extends Component {
  render() {
    const { recipeId } = this.props;
    const { recipe_name, recipe_description } = this.props.recipe;
    return (
      <div
        className="modal fade"
        id={recipeId}
        tabIndex="-1"
        role="dialog"
        aria-labelledby="exampleModalCenterTitle"
        aria-hidden="true"
      >
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title" id="exampleModalLongTitle">Procedure for preparing {recipe_name}</h5>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              <div className="procedure">
                {recipe_description}
              </div>
            </div>
            <div className="modal-footer">
              <button className="btn btn-danger" data-dismiss="modal">Close</button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Viewprocedure;
