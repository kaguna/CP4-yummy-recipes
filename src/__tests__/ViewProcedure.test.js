import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ViewProcedure from '../components/recipes/ViewProcedure';

describe('view procedure modal component', () => {
  const props = {
    recipe: { category_id: 1,
      recipe_description: 'fghbjnk',
      recipe_name: 'cool stuff' },
  };
  const wrapper = shallow(<ViewProcedure {...props} />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
