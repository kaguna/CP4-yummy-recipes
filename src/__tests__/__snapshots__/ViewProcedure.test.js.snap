// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view procedure modal component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal fade"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Procedure for preparing 
          cool stuff
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="procedure"
        >
          fghbjnk
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-danger"
          data-dismiss="modal"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
